xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-test/xwiki-platform-search-test-pageobjects/src/main/java/org/xwiki/search/test/po/QuickSearchElement.java

```diff
@@ -48,9 +48,19 @@ public void search(String terms) {
     /**
      * Get a suggest item from the "Page Titles" category.
      * @param index the index of the item to get
-     * @return the title of the item
+     * @return the title of the item, or null if it could not be found
      */
-    public WebElement getSuggestItemTitles(int index) {
-        return this.suggestItemsListTitles.findElements(By.xpath("//div[@class = 'value']")).get(index);
+    public WebElement getSuggestItemTitles(int index) throws InterruptedException
+    {
+        WebElement suggestItemsListTitles = this.getDriver().findElement(By.cssSelector(".results0 .suggestList"));
+        int nRetries = 5;
+        while (nRetries-- > 0) {
+            List<WebElement> titles = suggestItemsListTitles.findElements(By.xpath("//div[@class = 'value']"));
+            if (index < titles.size()) {
+                return titles.get(index);
+            }
+            Thread.sleep(500);
+        }
+        return null;
```

Comment:
I see there is a "loading" CSS class added to:

* the ``input#headerglobalsearchinput``,
* to ``li.showAllResults``
* to the individual ``div.results``

while searching. And there's also a "hidden" class added to the results sets that are empty. So I'm wondering if we couldn't rely on these instead of the ``Thread.sleep()`` which is an antipattern in functional tests. For instance, I'd try to:

* wait for ``.searchSuggest li.showAllResults.loading`` to be present (i.e. wait for a new search request to be sent)
* then wait for ``.searchSuggest .results0:not(.loading):not(.hidden)`` to be present (i.e. wait for the search results to be displayed)